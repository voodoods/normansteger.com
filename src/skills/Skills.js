import React, { Component } from 'react'
import PropTypes from 'prop-types'
import StyledSkills from './Skills.styled'
import SkillCard from './skill-card/SkillCard'
import ContentNotFound from '../content-not-found/ContentNotFound'
import Navbar from '../navbar/Navbar'

import { softSkills, codeSkills } from './skills-data.js'

class Skills extends Component {
  constructor(props) {
    super(props)

    this.state = {
      softSkills,
      codeSkills,
    }
  }

  render() {
    return (
      <StyledSkills className="Skills pure-u-lg-3-5 pure-u-md-1 pure-u-sm-1">
        {this.state.softSkills.length === 0 && <ContentNotFound />}

        <Navbar path={this.props.location.pathname} />

        {this.state.softSkills.length > 0 && (
          <div>
            <h3>
              <a name="professions">My Professions</a>
            </h3>
            {this.state.softSkills.map((skill, index) => {
              return <SkillCard key={index} logo={skill.logo} name={skill.name} description={skill.description} />
            })}
            <p />
            <br />
            <hr />
            <p />
            <h3>
              <a name="tools">Tools of Choice</a>
            </h3>
            {this.state.codeSkills.map((skill, index) => {
              return (
                <SkillCard
                  key={index}
                  logo={skill.logo}
                  name={skill.name}
                  description={skill.description}
                  centered={true}
                />
              )
            })}
          </div>
        )}
      </StyledSkills>
    )
  }
}

Skills.propTypes = {
  location: PropTypes.shape({
    pathname: PropTypes.string,
  }),
}

export default Skills
