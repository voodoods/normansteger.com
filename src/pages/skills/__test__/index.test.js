import React from 'react'
import { shallow } from 'enzyme'
import Skills from '../'

describe('<Skills /> component index', () => {
  it('renders without crashing', () => {
    const props = {
      location: {
        pathname: '/home/',
      },
      softSkills: [
        {
          logo: 'bla',
          name: 'bla bla',
          description: 'bla bla bla.',
        },
      ],
      codeSkills: [
        {
          logo: 'bla',
          name: 'bla bla',
          description: 'bla bla bla.',
        },
      ],
    }
    const wrapper = shallow(<Skills {...props} />)
    expect(wrapper).toMatchSnapshot()
  })
})
