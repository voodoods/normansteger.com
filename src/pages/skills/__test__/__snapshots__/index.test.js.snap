// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Skills /> component index renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <withProps(withProps(Skills))
    codeSkills={
        Array [
            Object {
              "description": "bla bla bla.",
              "logo": "bla",
              "name": "bla bla",
            },
          ]
    }
    location={
        Object {
            "pathname": "/home/",
          }
    }
    softSkills={
        Array [
            Object {
              "description": "bla bla bla.",
              "logo": "bla",
              "name": "bla bla",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "codeSkills": Array [
        Object {
          "description": "bla bla bla.",
          "logo": "bla",
          "name": "bla bla",
        },
      ],
      "location": Object {
        "pathname": "/home/",
      },
      "softSkills": Array [
        Object {
          "description": "I love building software for the web. Thus I always aim to use state-of-the-art technology while keeping my code as clean, fast and well tested as possbile.",
          "logo": "code.svg",
          "name": "I craft code",
        },
        Object {
          "description": "Passion for well designed user experiences and an eye for detail is part of my DNA. Being a trained media designer I turned this passion into a profession early on in my career.",
          "logo": "design.svg",
          "name": "I design media",
        },
        Object {
          "description": "Great results can only be achieved in a team. I've been part of such teams and have led them myself as wellc. Thus I'm always commited to provide my team mates with everything they need to achieve greatness - be it as their partner or their leader.",
          "logo": "team.svg",
          "name": "I'm a team player",
        },
      ],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "codeSkills": Array [
          Object {
            "description": "bla bla bla.",
            "logo": "bla",
            "name": "bla bla",
          },
        ],
        "location": Object {
          "pathname": "/home/",
        },
        "softSkills": Array [
          Object {
            "description": "I love building software for the web. Thus I always aim to use state-of-the-art technology while keeping my code as clean, fast and well tested as possbile.",
            "logo": "code.svg",
            "name": "I craft code",
          },
          Object {
            "description": "Passion for well designed user experiences and an eye for detail is part of my DNA. Being a trained media designer I turned this passion into a profession early on in my career.",
            "logo": "design.svg",
            "name": "I design media",
          },
          Object {
            "description": "Great results can only be achieved in a team. I've been part of such teams and have led them myself as wellc. Thus I'm always commited to provide my team mates with everything they need to achieve greatness - be it as their partner or their leader.",
            "logo": "team.svg",
            "name": "I'm a team player",
          },
        ],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
